app/dataDefinitions/defects.js

```diff
@@ -0,0 +1,10 @@
+const Defect = Symbol("Defect");
+
+export const isValid = object => object[Defect] === undefined;
+
+export const isFaulty = object => !isValid(object);
+
+export const getDefect = object => isValid(object) ? "" : object[Defect];
+
+export const NotFoundEntity = { [Defect]: "Not Found" };
```

Comment:
may be we shouldn't use entity for naming but data or object or .... i don t know something